datafusion/core/src/datasource/physical_plan/mod.rs

```diff
@@ -315,17 +313,18 @@ impl SchemaAdapter {
         }
 
         Ok((
-            SchemaMapping {
+            Arc::new(SchemaMapping {
                 table_schema: self.table_schema.clone(),
                 field_mappings,
-            },
+            }),
             projection,
         ))
     }
 }
 
 /// The SchemaMapping struct holds a mapping from the file schema to the table schema
 /// and any necessary type conversions that need to be applied.
+#[cfg(feature = "parquet")]
```

Comment:
It is unfortunate that we have to add the 

```
#[cfg(feature = "parquet")]
```
 all over the place

Maybe as a follow on PR we can pull this code into its own module (e.g. `datasource/schema_adaptor.rs` for example)